sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/implementation/util/ModelHelper.java

```diff
@@ -94,4 +109,90 @@ public static DataLakeAclChangeFailedException changeAclFailed(Exception e, Stri
+            }
+            if (options.getScheduleDeletionOptions() != null && options.getScheduleDeletionOptions().getTimeToExpire() != null) {
+                throw LOGGER.logExceptionAsError(new IllegalArgumentException("TimeToExpire does not apply to directories."));
+            }
+            if (options.getScheduleDeletionOptions() != null && options.getScheduleDeletionOptions().getExpiresOn() != null) {
+                throw LOGGER.logExceptionAsError(new IllegalArgumentException("ExpiresOn does not apply to directories."));
+            }
+        }
+        if (options.getScheduleDeletionOptions() == null) {
+            return null;
+        }
+        if (options.getScheduleDeletionOptions().getTimeToExpire() != null && options.getScheduleDeletionOptions().getExpiresOn() != null) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("TimeToExpire and ExpiresOn both cannot be set."));
+        }
+        if (options.getScheduleDeletionOptions().getTimeToExpire() != null) {
+            return PathExpiryOptions.RELATIVE_TO_NOW;
+        } else if (options.getScheduleDeletionOptions().getExpiresOn() != null) {
+            return PathExpiryOptions.ABSOLUTE;
+        }
+        return null;
+    }
+
+    /**
+     * Converts the metadata into a string of format "key1=value1, key2=value2" and Base64 encodes the values.
+     *
+     * @param metadata The metadata.
+     *
+     * @return The metadata represented as a String.
+     */
+    public static String buildMetadataString(Map<String, String> metadata) {
+        if (!CoreUtils.isNullOrEmpty(metadata)) {
+            StringBuilder sb = new StringBuilder();
+            for (final Map.Entry<String, String> entry : metadata.entrySet()) {
+                if (Objects.isNull(entry.getKey()) || entry.getKey().isEmpty()) {
+                    throw new IllegalArgumentException("The key for one of the metadata key-value pairs is null, "
+                        + "empty, or whitespace.");
+                } else if (Objects.isNull(entry.getValue()) || entry.getValue().isEmpty()) {
+                    throw new IllegalArgumentException("The value for one of the metadata key-value pairs is null, "
+                        + "empty, or whitespace.");
+                }
+
+                /*
+                The service has an internal base64 decode when metadata is copied from ADLS to Storage, so getMetadata
+                will work as normal. Doing this encoding for the customers preserves the existing behavior of
+                metadata.
+                 */
+                sb.append(entry.getKey()).append('=')
+                    .append(new String(Base64.getEncoder().encode(entry.getValue().getBytes(StandardCharsets.UTF_8)),
+                        StandardCharsets.UTF_8)).append(',');
+            }
+            sb.deleteCharAt(sb.length() - 1); // Remove the extraneous "," after the last element.
```

Comment:
FYI, `StringBuilder.deleteCharAt` may copy the entire buffer of the `StringBuilder`. So, if this is called often with a lot of metadata this could be doing unnecessary memory allocations. I'd prefer to use a pattern where we have a `boolean firstMetadata` and check that before appending the new key-value pair to append `,`.

```java
if (!firstMetadata) {
  sb.append(',');
}

sb.append(entry.getKey())
    .append('=')
    .append(new String(Base64.getEncoder().encode(entry.getValue().getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8));
firstMetadata = false;
```